packages/conduit-rxjs-react/package.json

```diff
@@ -32,17 +32,16 @@
   "bugs": "https://github.com/indiana-university/conduit/issues",
   "homepage": "https://github.com/indiana-university/conduit#readme",
   "peerDependencies": {
-    "conduit-rxjs": ">= 0.5.0",
-    "react": ">= 0.14.0",
+    "react": ">= 16.3.0",
```

Comment:
Since we know there's a top-range for compatibility, we can add that in here. I didn't find anything reliable that explicitly states it, but I'm imagining that `UNSAFE_componentWillReceiveProps` will be removed in v18?

```
"react": ">= 16.3.0 <18",
```

Reply:
All the information I had previously looked at made me think `componentWillReceiveProps` will be removed in React v17, but when testing updates to SER Cody discovered that it still existed. He found a blog post https://reactjs.org/blog/2020/10/20/react-v17.html that makes it sound like the breaking changes expected in v17 were postponed until v18. I expect v18 will now remove `componentWillReceiveProps` and throw warnings about `UNSAFE_componentWillReceiveProps`. This seems to follow from the React v17 code. There it has an automatic warning for using `componentWillReceiveProps`, but only warns about `UNSAFE_componentWillReceiveProps` in "strict mode". I expect in v18 `componentWillReceiveProps` will be cause an error and `UNSAFE_componentWillReceiveProps` will cause a warning whether you have "strict mode" enabled or not.

We could assume for now that we will not be compatible with v18 and update to say that we are if we find out that `UNSAFE_componentWillReceiveProps` will still work in v18.

Reply:
Investigating a little. [According to the blog](https://reactjs.org/blog/2021/06/08/the-plan-for-react-18.html#how-to-try-react-18-alpha-today):

> New alphas are regularly published to npm using the @alpha tag. These releases are built using the most recent commit to our main repo.

Searching the `main` branch (which should be the latest stuff going into v18), there are [some unit tests for `componentWillReceiveProps`](https://github.com/facebook/react/blob/cae635054e17a6f107a39d328649137b83f25972/packages/react/src/__tests__/createReactClassIntegration-test.js). These tests were last updated in August 2020. These tests still suggest that React invokes the legacy lifecycle methods under certain conditions. These tests don't seem to fully align yet with what I'd expect, given the stated depreciations.

Given this uncertainty, I revoke this suggestion. I don't want to put in the upper limit (`<18`) anymore. This particular Conduit version may break in some future version of React, but that's the general expectation for any software dependencies, anyway. And we will make an update soon enough to make this a null issue.

In summary — never mind.

Reply:
I didn't notice this message until after I put in the change. I'll remove the "< 18".